File: ios/Sources/ScanditCapacitorCore/Extensions.swift
/*
 * This file is part of the Scandit Data Capture SDK
 *
 * Copyright (C) 2023- Scandit AG. All rights reserved.
 */

import Foundation
import Capacitor

public extension NSDictionary {
    var jsonString: String {
        guard let theJSONData = try? JSONSerialization.data(withJSONObject: self,
                                                            options: []) else {
            return ""
        }

        return String(data: theJSONData, encoding: .utf8) ?? ""
    }
}

public extension Dictionary {
    var jsonString: String? {
        guard let data = try? JSONSerialization.data(withJSONObject: self, options: []) else {
            return nil
        }
        return String(data: data, encoding: .utf8)
    }
}

File: android/src/main/java/com/scandit/capacitor/datacapture/core/utils/Extensions.kt
/*
 * This file is part of the Scandit Data Capture SDK
 *
 * Copyright (C) 2019- Scandit AG. All rights reserved.
 */

package com.scandit.capacitor.datacapture.core.utils

import android.view.View
import android.view.ViewGroup
import com.scandit.datacapture.core.internal.sdk.AppAndroidEnvironment

fun Int.pxFromDp(): Float {
    val context = AppAndroidEnvironment.applicationContext
    val displayDensity = context.resources.displayMetrics.density
    return (this * displayDensity + 0.5f)
}

fun View.removeFromParent() {
    val parent = parent as? ViewGroup ?: return
    parent.removeView(this)
}
